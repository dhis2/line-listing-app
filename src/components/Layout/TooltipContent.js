import { ouIdHelper } from '@dhis2/analytics'
import i18n from '@dhis2/d2-i18n'
import PropTypes from 'prop-types'
import React from 'react'
import { connect } from 'react-redux'
import { getConditions } from '../../modules/conditions.js'
import {
    DIMENSION_TYPE_CATEGORY,
    DIMENSION_TYPE_CATEGORY_OPTION_GROUP_SET,
    DIMENSION_TYPE_ORGANISATION_UNIT_GROUP_SET,
    DIMENSION_TYPE_STATUS,
    DIMENSION_TYPE_PERIOD,
    DIMENSION_TYPE_OU,
} from '../../modules/dimensionConstants.js'
import { sGetMetadata } from '../../reducers/metadata.js'
import {
    sGetUiConditionsByDimension,
    sGetUiItemsByDimension,
} from '../../reducers/ui.js'
import styles from './styles/Tooltip.module.css'

const labels = {
    noneSelected: () => i18n.t('None selected'),
    allSelected: () => i18n.t('Showing all values for this dimension'),
    oneOverLimit: () => i18n.t('And 1 other...'),
    nOverLimit: (n) =>
        i18n.t('And {{numberOfItems}} others...', {
            numberOfItems: n,
        }),
}

const renderLimit = 5

export const TooltipContent = ({
    dimension,
    itemIds,
    metadata,
    conditions,
}) => {
    const getNameList = (idList, label, metadata) =>
        idList.reduce(
            (levelString, levelId, index) =>
                `${levelString}${index > 0 ? `, ` : ``}${
                    metadata[levelId] ? metadata[levelId].name : levelId
                }`,
            `${label}: `
        )

    const getItemDisplayNames = () => {
        const levelIds = []
        const groupIds = []
        const itemDisplayNames = []

        itemIds.forEach((id) => {
            if (ouIdHelper.hasLevelPrefix(id)) {
                levelIds.push(ouIdHelper.removePrefix(id))
            } else if (ouIdHelper.hasGroupPrefix(id)) {
                groupIds.push(ouIdHelper.removePrefix(id))
            } else {
                itemDisplayNames.push(metadata[id] ? metadata[id].name : id)
            }
        })

        levelIds.length &&
            itemDisplayNames.push(
                getNameList(levelIds, i18n.t('Levels'), metadata)
            )

        groupIds.length &&
            itemDisplayNames.push(
                getNameList(groupIds, i18n.t('Groups'), metadata)
            )

        return itemDisplayNames
    }

    const renderItems = (itemDisplayNames) => {
        const itemsToRender = itemDisplayNames
            .slice(0, renderLimit)
            .map((name) => (
                <li key={`${dimension.id}-${name}`} className={styles.item}>
                    {name}
                </li>
            ))

        if (itemDisplayNames.length > renderLimit) {
            itemsToRender.push(
                <li
                    key={`${dimension.id}-render-limit`}
                    className={styles.item}
                >
                    {itemDisplayNames.length - renderLimit === 1
                        ? labels.oneOverLimit
                        : labels.nOverLimit(
                              itemDisplayNames.length - renderLimit
                          )}
                </li>
            )
        }

        return itemsToRender
    }

    const renderConditions = () =>
        renderItems(getConditions({ conditions, metadata, dimension }))

    const renderNoItemsLabel = () => (
        <li
            key={`${dimension.id}-${labels.noneSelected()}`}
            className={styles.item}
        >
            {labels.noneSelected()}
        </li>
    )

    const renderAllItemsLabel = () => (
        <li
            key={`${dimension.id}-${labels.allSelected()}`}
            className={styles.item}
        >
            {labels.allSelected()}
        </li>
    )

    const itemDisplayNames = Boolean(itemIds.length) && getItemDisplayNames()

    switch (dimension.dimensionType) {
        case DIMENSION_TYPE_CATEGORY:
        case DIMENSION_TYPE_CATEGORY_OPTION_GROUP_SET:
        case DIMENSION_TYPE_ORGANISATION_UNIT_GROUP_SET:
        case DIMENSION_TYPE_STATUS:
            return (
                <ul className={styles.list}>
                    {itemDisplayNames
                        ? renderItems(itemDisplayNames)
                        : renderAllItemsLabel()}
                </ul>
            )

        case DIMENSION_TYPE_PERIOD:
        case DIMENSION_TYPE_OU:
            return (
                <ul className={styles.list}>
                    {itemDisplayNames
                        ? renderItems(itemDisplayNames)
                        : renderNoItemsLabel()}
                </ul>
            )

        default: {
            return (
                <ul className={styles.list}>
                    {conditions?.condition || conditions?.legendSet
                        ? renderConditions(conditions)
                        : renderAllItemsLabel()}
                </ul>
            )
        }
    }
}

TooltipContent.propTypes = {
    conditions: PropTypes.object.isRequired,
    dimension: PropTypes.object.isRequired,
    metadata: PropTypes.object.isRequired,
    itemIds: PropTypes.array,
}

const mapStateToProps = (state, ownProps) => ({
    metadata: sGetMetadata(state),
    itemIds: sGetUiItemsByDimension(state, ownProps.dimension.id) || [],
    conditions: sGetUiConditionsByDimension(state, ownProps.dimension.id) || {},
})

export default connect(mapStateToProps)(TooltipContent)
