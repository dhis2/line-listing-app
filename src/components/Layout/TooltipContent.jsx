import {
    DIMENSION_TYPE_CATEGORY,
    DIMENSION_TYPE_CATEGORY_OPTION_GROUP_SET,
    DIMENSION_TYPE_DATA_ELEMENT,
    ouIdHelper,
    DIMENSION_TYPE_ORGANISATION_UNIT,
    DIMENSION_TYPE_PERIOD,
    DIMENSION_TYPE_ORGANISATION_UNIT_GROUP_SET,
    AXIS_ID_FILTERS,
} from '@dhis2/analytics'
import i18n from '@dhis2/d2-i18n'
import PropTypes from 'prop-types'
import React from 'react'
import { useSelector } from 'react-redux'
import {
    isStartEndDate,
    useLocalizedStartEndDateFormatter,
} from '../../modules/dates.js'
import { DIMENSION_TYPE_STATUS } from '../../modules/dimensionConstants.js'
import { extractDimensionIdParts } from '../../modules/dimensionId.js'
import { sGetMetadata } from '../../reducers/metadata.js'
import { sGetUiInputType, sGetUiItemsByDimension } from '../../reducers/ui.js'
import styles from './styles/Tooltip.module.css'

const renderLimit = 5

export const TooltipContent = ({ dimension, conditionsTexts, axisId }) => {
    const metadata = useSelector(sGetMetadata)
    const inputType = useSelector(sGetUiInputType)
    const itemIds = useSelector((state) =>
        sGetUiItemsByDimension(state, dimension.id)
    )
    const formatStartEndDate = useLocalizedStartEndDateFormatter()

    const { programStageId, programId } = extractDimensionIdParts(
        dimension.id,
        inputType
    )

    const stageName = dimension.stageName || metadata[programStageId]?.name
    const programName = metadata[programId]?.name

    const getNameList = (idList, label, metadata) =>
        idList.reduce(
            (levelString, levelId, index) =>
                `${levelString}${index > 0 ? `, ` : ``}${
                    metadata[levelId] ? metadata[levelId].name : levelId
                }`,
            `${label}: `
        )

    const getItemDisplayNames = () => {
        const levelIds = []
        const groupIds = []
        const itemDisplayNames = []

        itemIds.forEach((id) => {
            if (ouIdHelper.hasLevelPrefix(id)) {
                levelIds.push(ouIdHelper.removePrefix(id))
            } else if (ouIdHelper.hasGroupPrefix(id)) {
                groupIds.push(ouIdHelper.removePrefix(id))
            } else {
                const { dimensionId } = extractDimensionIdParts(id, inputType)
                itemDisplayNames.push(
                    isStartEndDate(dimensionId)
                        ? formatStartEndDate(dimensionId)
                        : metadata[dimensionId]?.name ?? id
                )
            }
        })

        levelIds.length &&
            itemDisplayNames.push(
                getNameList(levelIds, i18n.t('Levels'), metadata)
            )

        groupIds.length &&
            itemDisplayNames.push(
                getNameList(groupIds, i18n.t('Groups'), metadata)
            )

        return itemDisplayNames
    }

    const renderItems = (itemDisplayNames = []) => {
        if (itemDisplayNames.some((name) => !name)) {
            return null
        }
        const itemsToRender = itemDisplayNames
            .slice(0, renderLimit)
            .map((name) => (
                <li key={`${dimension.id}-${name}`} className={styles.item}>
                    {name}
                </li>
            ))

        const numberOverRenderLimit = itemDisplayNames.length - renderLimit
        if (numberOverRenderLimit > 0) {
            itemsToRender.push(
                <li
                    key={`${dimension.id}-render-limit`}
                    className={styles.item}
                >
                    {i18n.t('And {{count}} other...', {
                        count: numberOverRenderLimit,
                        defaultValue: 'And {{count}} other...',
                        defaultValue_plural: 'And {{count}} others...',
                    })}
                </li>
            )
        }

        return itemsToRender
    }

    const renderNoItemsLabel = () => (
        <li key={`${dimension.id}-none-selected`} className={styles.item}>
            {i18n.t('None selected')}
        </li>
    )

    const renderStageName = () =>
        stageName && (
            <li className={styles.item}>
                {i18n.t('Program stage: {{- stageName}}', {
                    stageName,
                    nsSeparator: '^^',
                })}
            </li>
        )

    const renderProgramName = () =>
        programName && (
            <li className={styles.item}>
                {i18n.t('Program: {{- programName}}', {
                    programName,
                    nsSeparator: '^^',
                })}
            </li>
        )

    const renderItemsSection = (itemsList) => {
        if (itemsList.length) {
            return renderItems(itemsList)
        } else {
            if (axisId === AXIS_ID_FILTERS) {
                return renderNoItemsLabel()
            } else {
                return (
                    <li
                        key={`${dimension.id}-all-selected`}
                        className={styles.item}
                    >
                        {i18n.t('Showing all values for this dimension')}
                    </li>
                )
            }
        }
    }

    const itemDisplayNames = Boolean(itemIds.length) && getItemDisplayNames()

    switch (dimension.dimensionType) {
        case DIMENSION_TYPE_CATEGORY:
        case DIMENSION_TYPE_CATEGORY_OPTION_GROUP_SET:
        case DIMENSION_TYPE_ORGANISATION_UNIT_GROUP_SET:
        case DIMENSION_TYPE_STATUS:
            return (
                <ul className={styles.list} data-test="tooltip-content">
                    {renderProgramName()}
                    {renderItemsSection(itemDisplayNames)}
                </ul>
            )
        case DIMENSION_TYPE_PERIOD:
        case DIMENSION_TYPE_ORGANISATION_UNIT:
            return (
                <ul className={styles.list} data-test="tooltip-content">
                    {renderProgramName()}
                    {itemDisplayNames
                        ? renderItems(itemDisplayNames)
                        : renderNoItemsLabel()}
                </ul>
            )
        case DIMENSION_TYPE_DATA_ELEMENT: {
            return (
                <ul className={styles.list} data-test="tooltip-content">
                    {renderProgramName()}
                    {renderStageName()}
                    {renderItemsSection(conditionsTexts)}
                </ul>
            )
        }
        default: {
            return (
                <ul className={styles.list} data-test="tooltip-content">
                    {renderProgramName()}
                    {renderItemsSection(conditionsTexts)}
                </ul>
            )
        }
    }
}

TooltipContent.propTypes = {
    conditionsTexts: PropTypes.array.isRequired,
    dimension: PropTypes.object.isRequired,
    axisId: PropTypes.string,
}

export default TooltipContent
