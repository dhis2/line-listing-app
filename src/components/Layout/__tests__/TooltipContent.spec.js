import { render } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import { TooltipContent } from '../TooltipContent.js'

const mockStore = configureMockStore()

describe('TooltipContent', () => {
    test('ou and level', () => {
        const store = {
            ui: {
                itemsByDimension: {
                    ou: ['equestriaOuId', 'LEVEL-evertreeForestId'],
                },
                conditions: {},
            },
            metadata: {
                equestriaOuId: { name: 'OU Equestria' },
                evertreeForestId: { name: 'Evertree forest' },
            },
        }

        const dimension = {
            id: 'ou',
            dimensionType: 'ORGANISATION_UNIT',
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <TooltipContent dimension={dimension} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
    test('ou with missing metadata', () => {
        const store = {
            ui: {
                itemsByDimension: {
                    ou: ['equestriaOuId', 'LEVEL-evertreeForestId'],
                },
                conditions: {},
            },
            metadata: {},
        }

        const dimension = {
            id: 'ou',
            dimensionType: 'ORGANISATION_UNIT',
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <TooltipContent dimension={dimension} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
    test('ou id and 3 ou groups', () => {
        const store = {
            ui: {
                itemsByDimension: {
                    ou: [
                        'equestriaOuId',
                        'OU_GROUP-bluePonies',
                        'OU_GROUP-yellowPonies',
                        'OU_GROUP-purplePonies',
                    ],
                },
                conditions: {},
            },
            metadata: {
                equestriaOuId: { name: 'OU Equestria' },
                bluePonies: { name: 'All the blue ponies' },
                yellowPonies: { name: 'All the yellow ponies' },
                purplePonies: { name: 'All the purple ponies' },
            },
        }

        const dimension = {
            id: 'ou',
            dimensionType: 'ORGANISATION_UNIT',
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <TooltipContent dimension={dimension} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
    test('no conditions', () => {
        const store = {
            ui: { itemsByDimension: {}, conditions: {} },
            metadata: {},
        }

        const dimension = {
            id: 'dataElementId',
            dimensionType: 'DATA_ELEMENT',
            valueType: 'NUMBER',
            optionSet: undefined,
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <TooltipContent dimension={dimension} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
    test('COGS', () => {
        const store = {
            ui: {
                itemsByDimension: {
                    COGS_id: ['cogs1Id', 'cogs2Id'],
                },
                conditions: {},
            },
            metadata: {
                cogs1Id: {
                    id: 'cogs1Id',
                    name: 'COGS item 1',
                    code: 'cogs1Code',
                },
                cogs2Id: {
                    id: 'cogs2Id',
                    name: 'COGS item 2',
                    code: 'cogs2Code',
                },
            },
        }

        const dimension = {
            id: 'COGS_id',
            dimensionType: 'CATEGORY_OPTION_GROUP_SET',
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <TooltipContent dimension={dimension} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
    test('no period selected', () => {
        const store = {
            ui: { itemsByDimension: {}, conditions: {} },
            metadata: {
                eventDate: { name: 'Report compilation date' },
            },
        }

        const dimension = {
            id: 'eventDate',
            dimensionType: 'PERIOD',
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <TooltipContent dimension={dimension} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
    test('period selected', () => {
        const store = {
            ui: {
                itemsByDimension: {
                    eventDate: ['202205', '202207'],
                },
                conditions: {},
            },
            metadata: {
                202205: {
                    id: '202205',
                    name: 'May 2022',
                },
                202207: {
                    id: '202207',
                    name: 'July 2022',
                },
            },
        }

        const dimension = {
            id: 'eventDate',
            dimensionType: 'PERIOD',
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <TooltipContent dimension={dimension} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
    test('no ou selected', () => {
        const store = {
            ui: { itemsByDimension: {}, conditions: {} },
            metadata: {},
        }

        const dimension = {
            id: 'ou',
            dimensionType: 'ORGANISATION_UNIT',
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <TooltipContent dimension={dimension} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
    test('return null when at least one item is missing a name', () => {
        const store = {
            ui: {
                itemsByDimension: {},
                conditions: {
                    myDataElementId: {
                        condition: 'IN:ModernaVaxCode;PfizerVaxCode',
                    },
                },
            },
            metadata: {
                PfizerVaxId: {
                    id: 'PfizerVaxId',
                    name: 'The Pfizer Vaccine',
                    code: 'PfizerVaxCode',
                },
            },
        }

        const dimension = {
            id: 'myDataElementId',
            name: 'My fascinating Data Element',
            dimensionType: 'DATA_ELEMENT',
            valueType: 'TEXT',
            optionSet: 'optionSetId',
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <TooltipContent dimension={dimension} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
    test('with stage', () => {
        const store = {
            ui: {
                itemsByDimension: {},
                conditions: {
                    'stageId.dataElementId': {
                        condition: 'GT:11',
                    },
                },
            },
            metadata: {
                stageId: { name: 'The first stage' },
                'stageId.dataElementId': { name: 'Lucky numbers' },
            },
        }

        const dimension = {
            id: 'stageId.dataElementId',
            dimensionType: 'DATA_ELEMENT',
            valueType: 'NUMBER',
            name: 'Lucky numbers',
            optionSet: undefined,
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <TooltipContent dimension={dimension} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
    test('legendset without options chosen', () => {
        const store = {
            ui: {
                itemsByDimension: {},
                conditions: {
                    dimensionId: {
                        condition: '',
                        legendSet: 'legendSetId',
                    },
                },
            },
            metadata: {
                legendSetId: {
                    id: 'legendSetId',
                    legends: [
                        {
                            startValue: 0,
                            endValue: 5,
                            id: 'legendOption1Id',
                            name: '0 - 4',
                        },
                        {
                            startValue: 5,
                            endValue: 15,
                            id: 'legendOption2Id',
                            name: '5 - 14',
                        },
                        {
                            startValue: 15,
                            endValue: 25,
                            id: 'evLlhbRsG6e',
                            name: '15 - 24',
                        },
                    ],
                    name: 'Legend set name',
                },
            },
        }

        const dimension = {
            id: 'dimensionId',
            dimensionType: 'DATA_ELEMENT',
            valueType: 'NUMBER',
            legendSets: [{ id: 'legendSetId', name: 'Legend set name' }],
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <TooltipContent dimension={dimension} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
    test('legendset with options chosen', () => {
        const store = {
            ui: {
                itemsByDimension: {},
                conditions: {
                    dimensionId: {
                        condition: 'IN:legendOption1Id;legendOption2Id',
                        legendSet: 'legendSetId',
                    },
                },
            },
            metadata: {
                legendSetId: {
                    id: 'legendSetId',
                    legends: [
                        {
                            startValue: 0,
                            endValue: 5,
                            id: 'legendOption1Id',
                            name: '0 - 4',
                        },
                        {
                            startValue: 5,
                            endValue: 15,
                            id: 'legendOption2Id',
                            name: '5 - 14',
                        },
                        {
                            startValue: 15,
                            endValue: 25,
                            id: 'evLlhbRsG6e',
                            name: '15 - 24',
                        },
                    ],
                    name: 'Legend set name',
                },
            },
        }

        const dimension = {
            id: 'dimensionId',
            dimensionType: 'DATA_ELEMENT',
            valueType: 'NUMBER',
            legendSets: [{ id: 'legendSetId', name: 'Legend set name' }],
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <TooltipContent dimension={dimension} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
    test('2 conditions', () => {
        const store = {
            ui: {
                itemsByDimension: {},
                conditions: {
                    dimensionId: {
                        condition: 'NE:NV:LT:1000',
                    },
                },
            },
            metadata: {},
        }

        const dimension = {
            id: 'dimensionId',
            dimensionType: 'DATA_ELEMENT',
            valueType: 'NUMBER',
            optionSet: undefined,
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <TooltipContent dimension={dimension} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
    test('6 conditions', () => {
        const store = {
            ui: {
                itemsByDimension: {},
                conditions: {
                    dimensionId: {
                        condition: 'NE:NV:LT:1000:GT:10:!EQ:11:!EQ:50:!EQ:35',
                    },
                },
            },
            metadata: {},
        }

        const dimension = {
            id: 'dimensionId',
            dimensionType: 'DATA_ELEMENT',
            valueType: 'NUMBER',
            optionSet: undefined,
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <TooltipContent dimension={dimension} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
    // TODO: figure out why plural translations are not being returned from i18n.t() in Jest tests
    test.skip('8 conditions', () => {
        const store = {
            ui: {
                itemsByDimension: {},
                conditions: {
                    dimensionId: {
                        condition:
                            'NE:NV:LT:1000:GT:10:!EQ:11:!EQ:50:!EQ:35:!EQ:3500:EQ:33',
                    },
                },
            },
            metadata: {},
        }

        const dimension = {
            id: 'dimensionId',
            dimensionType: 'DATA_ELEMENT',
            valueType: 'NUMBER',
            optionSet: undefined,
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <TooltipContent dimension={dimension} />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
})
