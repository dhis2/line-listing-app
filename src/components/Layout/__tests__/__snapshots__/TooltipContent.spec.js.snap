// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipContent 2 conditions 1`] = `
<div>
  <ul
    class="list"
  >
    <li
      class="item"
    >
      Is not empty / not null
    </li>
    <li
      class="item"
    >
      Less than (&lt;): 1000
    </li>
  </ul>
</div>
`;

exports[`TooltipContent 6 conditions 1`] = `
<div>
  <ul
    class="list"
  >
    <li
      class="item"
    >
      Is not empty / not null
    </li>
    <li
      class="item"
    >
      Less than (&lt;): 1000
    </li>
    <li
      class="item"
    >
      Greater than (&gt;): 10
    </li>
    <li
      class="item"
    >
      Not equal to (≠): 11
    </li>
    <li
      class="item"
    >
      Not equal to (≠): 50
    </li>
    <li
      class="item"
    >
      And 1 other...
    </li>
  </ul>
</div>
`;

exports[`TooltipContent 8 conditions 1`] = `
<div>
  <ul
    class="list"
  >
    <li
      class="item"
    >
      Is not empty / not null
    </li>
    <li
      class="item"
    >
      Less than (&lt;): 1000
    </li>
    <li
      class="item"
    >
      Greater than (&gt;): 10
    </li>
    <li
      class="item"
    >
      Not equal to (≠): 11
    </li>
    <li
      class="item"
    >
      Not equal to (≠): 50
    </li>
    <li
      class="item"
    >
      And 3 others...
    </li>
  </ul>
</div>
`;

exports[`TooltipContent COGS 1`] = `
<div>
  <ul
    class="list"
  >
    <li
      class="item"
    >
      COGS item 1
    </li>
    <li
      class="item"
    >
      COGS item 2
    </li>
  </ul>
</div>
`;

exports[`TooltipContent legendset with options chosen 1`] = `
<div>
  <ul
    class="list"
  >
    <li
      class="item"
    >
      0 - 4
    </li>
    <li
      class="item"
    >
      5 - 14
    </li>
  </ul>
</div>
`;

exports[`TooltipContent legendset without options chosen 1`] = `
<div>
  <ul
    class="list"
  >
    <li
      class="item"
    >
      Legend set name
    </li>
  </ul>
</div>
`;

exports[`TooltipContent no conditions 1`] = `
<div>
  <ul
    class="list"
  >
    <li
      class="item"
    >
      Showing all values for this dimension
    </li>
  </ul>
</div>
`;

exports[`TooltipContent no ou selected 1`] = `
<div>
  <ul
    class="list"
  >
    <li
      class="item"
    >
      None selected
    </li>
  </ul>
</div>
`;

exports[`TooltipContent no period selected 1`] = `
<div>
  <ul
    class="list"
  >
    <li
      class="item"
    >
      None selected
    </li>
  </ul>
</div>
`;

exports[`TooltipContent ou and level 1`] = `
<div>
  <ul
    class="list"
  >
    <li
      class="item"
    >
      OU Equestria
    </li>
    <li
      class="item"
    >
      Levels: Evertree forest
    </li>
  </ul>
</div>
`;

exports[`TooltipContent ou id and 3 ou groups 1`] = `
<div>
  <ul
    class="list"
  >
    <li
      class="item"
    >
      OU Equestria
    </li>
    <li
      class="item"
    >
      Groups: All the blue ponies, All the yellow ponies, All the purple ponies
    </li>
  </ul>
</div>
`;

exports[`TooltipContent ou with missing metadata 1`] = `
<div>
  <ul
    class="list"
  >
    <li
      class="item"
    >
      equestriaOuId
    </li>
    <li
      class="item"
    >
      Levels: evertreeForestId
    </li>
  </ul>
</div>
`;

exports[`TooltipContent period selected 1`] = `
<div>
  <ul
    class="list"
  >
    <li
      class="item"
    >
      May 2022
    </li>
    <li
      class="item"
    >
      July 2022
    </li>
  </ul>
</div>
`;

exports[`TooltipContent return null when at least one item is missing a name 1`] = `
<div>
  <ul
    class="list"
  />
</div>
`;

exports[`TooltipContent with stage 1`] = `
<div>
  <ul
    class="list"
  >
    <li
      class="item"
    >
      Program stage: The first stage
    </li>
    <li
      class="item"
    >
      Greater than (&gt;): 11
    </li>
  </ul>
</div>
`;
